### DISCUSSION TITLE
Kafka scaler trigger on multiple kafka topics for scaling the same deployment

### DISCUSSION THREAD

**Discussion by anandsoni11** [kedacore/keda#4084] [Q&A / Need Help] [answered]
When we define the ScaledObject in KEDA, can we have triggers on multiple kafka topics with separate thresholds? KEDA should scale the deployment if any of the trigger conditions is met. If this can be done -
- How are the conflicts handled between the triggers (one trigger within threshold triggering HPA scale down and another trigger breaching threshold triggering scale up)?
- How is this behavior defined in the ScaledObject? Some samples will help a lot.

I have looked at https://github.com/kedacore/keda/issues/828 this issue but I am not clear if this is supported or not.

Thank you!

**Comment by zroubalik** [comment, marked as answer]
Yes, you can have mutliple triggers. Each trigger could be even a different type (ie. Prom + Kafka + ....).

All values are being sent to HPA and it does the decision on scaling - the greatest value  out of all scalers is selected for the scaling target.
For activation  (0-1 scaling), once one scaler is active, the whole ScaledObject is active.

You should not use multiple ScaledObjects/HPAs targeting the same resource, it will cause problems.
